subxt/src/client/online_client.rs

```diff
@@ -209,34 +234,119 @@ impl<T: Config> ClientRuntimeUpdater<T> {
         &curr.runtime_version != new
     }
 
+    fn do_update(&self, update: Update) {
+        let mut writable = self.0.inner.write();
+        writable.metadata = update.metadata;
+        writable.runtime_version = update.runtime_version;
+    }
+
+    /// Tries to apply a new update.
+    pub fn apply_update(&self, update: Update) -> Result<(), UpgradeError> {
+        if !self.is_runtime_version_different(&update.runtime_version) {
+            return Err(UpgradeError::SameVersion)
+        }
+
+        self.do_update(update);
+
+        Ok(())
+    }
+
     /// Performs runtime updates indefinitely unless encountering an error.
     ///
     /// *Note:* This will run indefinitely until it errors, so the typical usage
     /// would be to run it in a separate background task.
     pub async fn perform_runtime_updates(&self) -> Result<(), Error> {
         // Obtain an update subscription to further detect changes in the runtime version of the node.
-        let mut update_subscription = self.0.rpc.subscribe_runtime_version().await?;
+        let mut runtime_version_stream = self.0.rpc().subscribe_runtime_version().await?;
```

Comment:
I guess you could scrap this function body and swap it with something like:

```rust
let mut update_stream = updater.runtime_updates().await.unwrap();

while let Some(Ok(update)) = update_stream.next().await {
    let version = update.runtime_version().spec_version;

    if let Err(e) = updater.apply_update(update) {
        return Err(e)
    };
}
```

So that the actual update logic is all in one place, just using the low level bits you've added :)